base/Dockerfile

```diff
@@ -12,10 +12,14 @@ RUN conda install --yes \
     nomkl \
     numpy==1.16.2 \
     pandas==0.24.2 \
-    && conda clean -tipsy
+    && conda update conda -y \
+    && conda clean -tipsy \
+    && find /opt/conda/ -type f,l -name '*.a' -delete \
+    && find /opt/conda/ -type f,l -name '*.pyc' -delete \
+    && find /opt/conda/ -type f,l -name '*.js.map' -delete \
+    && rm -rf /opt/conda/pkgs
```

Comment:
Am a little wary of this last `rm` line. The tarballs should have already been cleaned by `conda clean -t`. The rest of this content here should be unpacked tarballs, which are actively in use. I don't think we should remove those.

Reply:
They're hardlinks, so the actual files are still there. I was surprised that this reduced the image size - I wonder if when docker compresses a layer it doesn't catch that hardlinks are identical and thus duplicates the data (the savings here are non-negligible). Conda is robust to a missing package cache, so I don't see this being a problem at runtime (I also tested installing/upgrading packages at runtime and things seemed to work).

Reply:
Yep, they are hard-linked. So things may work ok.

Am also surprised to hear this decreased image size. I wonder to what extent the union filesystem used impacts and to what extent Docker itself does. FWIW in my search to answer these questions, I came across this old PR ( https://github.com/moby/moby/pull/16960 ).

Reply:
There has been some discussion about this in Jupyter docker-stacks. In particular, [this comment]( https://github.com/jupyter/docker-stacks/issues/861#issuecomment-489737844 ) seems relevant.